### DISCUSSION TITLE
reference_classif.txt

### DISCUSSION THREAD

**Discussion by pawel-wojciechowski** [ConesaLab/SQANTI3#285] [Q&A] [answered]
Hi,
does anyone know what "The path to the reference transcriptome classification file is /path/to/reference_classif.txt" means . Which file is it exactly? It appears as a parameter of the SQANTI3 rescue step.

Pawel

**Comment by carolinamonzo** [comment, marked as answer]
Hi @pawel-wojciechowski it's explained in the SQANTI wiki / Running SQANTI3 rescue / SQANTI3 resque strategy / 3. Application of SQ3 filter to the reference transcriptome. You need to generate the reference_classif.txt file by running SQANTI3 QC on your reference transcriptome, using also the same orthogonal data you used on your samples.

I hope this solves your question!
Best,
Carolina.

**Comment by pawel-wojciechowski** [comment]
Hi Caroline,
indeed everything is described. Sorry for the confusion. I would recommend possibly adding a link to the Wiki in the https://github.com/ConesaLab/SQANTI3/wiki/Tutorial:-running-SQANTI3-on-an-example-dataset#Executing-and-Understanding-SQANTI3-Rescue section to avoid further questions (it's not obvious in the video tutorial). I don't know why but somehow I found it to be the most accurate description. 
In any case, thank you! 

Great job!
Best regards,
Pawel.

**Reply by carolinamonzo** [reply]
Great! I'll do that! Thanks a lot!
Best wishes,
Carolina.

**Reply by mpizzagalli777** [reply]
Sorry I have a quick follow up question to this. When running the reference transcriptome, is it simply as follows? I was a little unclear on how to run QC on the transcriptome correctly. 
```
python $sqanti/sqanti3_qc.py $genome/gencode.v44.annotation.gtf $genome/gencode.v44.annotation.gtf $genome/GRCh38.p14.genome.fa \
	-c $sr_dir/G2_1_filtered_SJ.out.tab,$sr_dir/G2_2_filtered_SJ.out.tab \
	--SR_bam $wkdir/bam_list.txt \
	-o gencode.v44.annotation -d SQANTI3_output --report skip --force_id_ignore
```
Thanks for the help!

**Reply by carolinamonzo** [reply]
Hi, yes, exactly, you simply run it with the reference genome GTF as input (like as if you were running the GTF of a sample of interest). After this, when you run sqanti3_resque, remember to add the output *_classification.txt to --refClassif.

**Reply by MR-D-CJ** [reply]
To generate the `reference_classif.txt` file, when I run `sqanti3_qc.py` against reference, my input data is the reference transcript data, something like the following, is this correct?

`python sqanti3_qc.py \
/PATH/GRCh38/genecode/gencode.v47.transcripts.fa \
/PATH/GRCh38/genecode/gencode.v46.annotation.gtf \
/PATH/GRCh38/GCA_000001405.15_GRCh38_no_alt/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa \
--CAGE_peak /PATH/human.refTSS_v3.1.hg38.sorted.bed \
--polyA_motif_list /PATH/polyA_info/mouse_and_human.polyA_motif.txt \
-o GRCh38 \
-d /PATH/SQANTi3_ref/GRCh38_QC \
--fasta \
--force_id_ignore \
--cpus 20 --report html`

**Reply by carolinamonzo** [reply]
Hi, yes, sorry, I forgot to answer

**Reply by MR-D-CJ** [reply]
> Hi, yes, sorry, I forgot to answer

Thank u！

**Comment by MR-D-CJ** [comment]
sorry,I have another problem.I run sqanti.qc for collapsed file. when i use refTSS_v4.1_human_coordinate.hg38.bed.txt from https://reftss.riken.jp/datafiles/4.1/human/,
can't create *_classification.txt, i don't know why?
code is 
`python /share/home/duchangjian/soft/biosoft/SQANTI3-5.2.2/sqanti3_qc.py \
/PATH/Sample.GRCh38-no-alt.collapsed.gff \
/PATH/human/GRCh38/genecode/gencode.v46.annotation.gtf \
/PATH/human/GRCh38/GCA_000001405.15_GRCh38_no_alt/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa \
--CAGE_peak /PATH/refTSS/human.refTSS_v4.1.hg38.sorted.bed \
--polyA_motif_list /PATH/polyA_info/mouse_and_human.polyA_motif.txt \
-o SampleGRCh38 \
-d /PATH/sqanti.refTSS4 \
-fl /PATH/Sample.GRCh38-no-alt.collapsed.abundance.txt \
--cpus 20 --report html 2>>"$LOGFILE"`

ouptput file only have:
<img width="1582" alt="截屏2024-12-19 00 42 36" src="https://github.com/user-attachments/assets/61118cbc-6796-48d3-ae55-4e205642aa67" />

log file show that:
<img width="1912" alt="截屏2024-12-19 00 15 48" src="https://github.com/user-attachments/assets/38667f6e-5f7d-422c-ab3f-cf4f37b251c1" />

refTSS_v4.1_human_coordinate.hg38.bed.txt file is 
<img width="729" alt="截屏2024-12-19 00 46 20" src="https://github.com/user-attachments/assets/f2705c2a-327b-4bb6-ab07-1ce4f8eeab09" />
